pkg/cmd/server/start/start_node.go

```diff
@@ -225,17 +226,17 @@ func (o NodeOptions) CreateNodeConfig() error {
 	return nil
 }
 
-func RunSDNController(config configapi.NodeConfig) {
+func RunSDNController(config configapi.NodeConfig, nodeConfig *kubernetes.NodeConfig) {
 	if config.NetworkPluginName == osdn.NetworkPluginName() {
-		kclient, _, err := configapi.GetKubeClient(config.MasterKubeConfig)
-		if err != nil {
-			glog.Fatal("Failed to get kube client for SDN")
-		}
 		oclient, _, err := configapi.GetOpenShiftClient(config.MasterKubeConfig)
 		if err != nil {
 			glog.Fatal("Failed to get kube client for SDN")
 		}
-		osdn.Node(*oclient, *kclient, config.NodeName, "")
+		ch := make(chan struct{})
+		nodeConfig.KubeletConfig.StartUpdates = ch
+		go osdn.Node(oclient, nodeConfig.Client, config.NodeName, "", ch)
+		// no plugin necessary
+		nodeConfig.KubeletConfig.NetworkPluginName = ""
```

Comment:
The Kubelet does not have your plugin registered and so it fails to start.  My understanding was that you were programming the bridge so you didn't need a plugin?  Where is your plugin code registered.

----- Original Message -----
> >  		oclient, _, err := configapi.GetOpenShiftClient(config.MasterKubeConfig)
> >  		if err != nil {
> >  			glog.Fatal("Failed to get kube client for SDN")
> >  		}
> > -		osdn.Node(*oclient, *kclient, config.NodeName, "")
> > +		ch := make(chan struct{})
> > +		nodeConfig.KubeletConfig.StartUpdates = ch
> > +		go osdn.Node(oclient, nodeConfig.Client, config.NodeName, "", ch)
> > +		// no plugin necessary
> > +		nodeConfig.KubeletConfig.NetworkPluginName = ""
> 
> This line is wrong.. it breaks the SDN. Why do you think we do not need the
> plugin now?
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/openshift/origin/pull/2984/files#r32058218